### DISCUSSION TITLE
Compilation takes a long time

### DISCUSSION THREAD

**Discussion by alexccc99** [ClickHouse/ClickHouse#31475] [Q&A] [answered]
When I execute "cmake --build . --config RelWithDebInfo", it takes a long time, this means that even if I only change a little code, I need to recompile for a long time to verify my ideas. Do you usually use this development method?

**Comment by filimonov** [comment, marked as answer]
Please follow the official recommendation about the build.
https://clickhouse.com/docs/en/development/build/#normal-build-for-development-on-ubuntu

Normally you do `cmake` call once, after that you run `ninja`. The first build is slow (about ~40min on a modern laptop to 2-3 hours on old hw). Next rebuilds (just run ninja) should be much faster. 

If you need to speed up the build for development purposes you can additionally change the linking strategy https://clickhouse.com/docs/en/development/build/#faster-builds-for-development or disable unneeded modules.

**Reply by alexccc99** [reply]
I'm building clickhouse on macos. Execute the following command：
```
cd ClickHouse
rm -rf build
mkdir build
cd build
cmake -DCMAKE_C_COMPILER=$(brew --prefix llvm)/bin/clang -DCMAKE_CXX_COMPILER=$(brew --prefix llvm)/bin/clang++ -DCMAKE_BUILD_TYPE=RelWithDebInfo ..
cmake --build . --config RelWithDebInfo
cd ..
```
It has no "nijia" command. So how should I rebuild faster?
